File: 1791-find-center-of-star-graph/1791-find-center-of-star-graph.cpp
class Solution {
public:
    int findCenter(vector<vector<int>>& edges) {
        unordered_map<int, int> nodeCount;
        for (const auto& edge : edges) {
            nodeCount[edge[0]]++;
            nodeCount[edge[1]]++;
        }

        for (const auto& [node, count] : nodeCount) {
            if (count == edges.size()) {
                return node;
            }
        }
        return -1;
    }
};

File: 1791-find-center-of-star-graph/1791-find-center-of-star-graph.cs
public class Solution {
    public int FindCenter(int[][] edges) {
        Dictionary<int, int> nodeCount = new Dictionary<int, int>();
        
        foreach (var edge in edges)
        {
            if (!nodeCount.ContainsKey(edge[0]))
            {
                nodeCount[edge[0]] = 0;
            }
            if (!nodeCount.ContainsKey(edge[1]))
            {
                nodeCount[edge[1]] = 0;
            }

            nodeCount[edge[0]]++;
            nodeCount[edge[1]]++;
        }

        foreach (var kvp in nodeCount)
        {
            if (kvp.Value == edges.Length)
            {
                return kvp.Key;
            }
        }

        return -1; 
    }
}